src/main/java/org/marketdesignresearch/mechlib/core/bid/demand/DemandBids.java

```diff
@@ -0,0 +1,52 @@
+package org.marketdesignresearch.mechlib.core.bid.demand;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import org.marketdesignresearch.mechlib.core.Good;
+import org.marketdesignresearch.mechlib.core.bid.Bids;
+import org.marketdesignresearch.mechlib.core.bid.bundle.BundleValueBid;
+import org.marketdesignresearch.mechlib.core.bid.bundle.BundleValueBids;
+import org.marketdesignresearch.mechlib.core.bid.bundle.BundleValuePair;
+import org.marketdesignresearch.mechlib.core.bidder.Bidder;
+import org.marketdesignresearch.mechlib.core.price.Prices;
+
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+public class DemandBids extends Bids<DemandBid> {
+
+	public DemandBids(Map<Bidder, DemandBid> bidderBidMap) {
+		super(bidderBidMap);
+	}
+
+	public DemandBids() {
+		super();
+	}
+
+	@Override
+	protected DemandBid createEmptyBid() {
+		return null;

```

Comment:
Shouldn't that rather return an empty demand bid, or something similar?

Reply:
The question is what is an empty bid? Is it where the bidder did specify to submit no bid or is it that the bidder shows his demand for the empty bundle? I think the first one is an empty bid to me, but of course you can argue for both.

Reply:
Just a little side note on this: I just struggled over the methodBids::getBid which I would as well define different. I.e. If a bidder did not submit a bid is different from a bidder that submit an empty bid.

Reply:
Good points, I don't have a strong opinion. I'm rather thinking of an empty bid as a "neutral" bid, as if the bidder didn't submit a bid at all... Or in other words, a bid that doesn't provide any more information than what the auctioneer know before already.

Reply:
I actually changed this method to return a demand bid on the empty bundle. Technically this is a better solution than returning null. And otherwise I do not have a strong opinion on this.

Reply:
👍